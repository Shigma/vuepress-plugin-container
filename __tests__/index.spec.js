const { fs, path } = require('@vuepress/shared-utils')
const { createApp } = require('@vuepress/core')
const ContainerPlugin = require('..')

describe('containers', () => {
  let app

  beforeAll(async () => {
    app = createApp({
      plugins: [
        [ContainerPlugin, { type: 'tip' }],
      ],
    })
    return app.process()
  })

  const fragmentDir = path.join(__dirname, 'fragments')
  fs.readdirSync(fragmentDir).forEach((name) => {
    const filepath = path.join(fragmentDir, name)
    const content = fs.readFileSync(filepath, 'utf8')
    test(name, () => {
      const { html } = app.markdown.render(content)
      expect(html).toMatchSnapshot()
    })
  })
})
