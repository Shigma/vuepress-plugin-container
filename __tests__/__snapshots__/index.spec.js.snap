// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers tip.md 1`] = `
<div class="tip custom-block">
  <p class="custom-block-title">TIP</p>
  <p>I am a tip</p>
</div>
`;

exports[`containers tip-override.md 1`] = `
<div class="tip custom-block">
  <p class="custom-block-title">提示</p>
  <p>I am a tip</p>
</div>
`;
